### PR TITLE
add atp attestation mappings to enroll

### PR DESCRIPTION
**Pull request by yosefserkez**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [x] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: TT6-185610738

# A brief description of the changes

Current behavior:
The fields outlined in the ticket are missing from the application contract, preventing them from being allowed as parameters.

New behavior:
The fields are added to the contract enabling them to successfully be passed as parameters.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

I have also updated the error output to use the full message since just the error object is unhelpful.

### CODE REVIEW

**Review by erubinst — DISMISSED** [review]
looks good to me.  Adds years to renew and attestation mappings, fixes spelling and logging

File: components/financial_assistance/app/domain/financial_assistance/operations/transfers/medicaid_gateway/account_transfer_in.rb [hunk 1]
```diff
@@ -185,7 +185,9 @@ def build_applicants(payload, application, family)
           end
 
           def build_application(payload, family)
-            app_params = payload["family"]['magi_medicaid_applications'].first.merge!(family_id: family.id, benchmark_product_id: BSON::ObjectId.new, years_to_renew: 5)
+            app = payload["family"]['magi_medicaid_applications'].first
+            # years_to_renew needs to be merged with a hash rocket to properly merge with the existing years_to_renew key
+            app_params = app.merge!(family_id: family.id, benchmark_product_id: BSON::ObjectId.new, "years_to_renew" => 5)
```
[the remark this answers is not in the document]
Comment (erubinst) [reply components/financial_assistance/app/domain/financial_assistance/operations/transfers/medicaid_gateway/account_transfer_in.rb:190]: Can add to account_transfer_in_spec and update the xml there - I created some tests in here for immigration documents that would be similar.  If you run into any trouble where things pass locally but not remotely, check that the associated dummy files are up to date

File: components/financial_assistance/spec/shared_examples/medicaid_gateway/Simple_Test_Case_E_New.xml [hunk 2]
```diff
@@ -633,4 +634,4 @@
         <ns5:HouseholdMemberReference ns1:ref="pe1992374604681766995"/>
         <ns5:HouseholdMemberReference ns1:ref="pe1992374604681766996"/>
     </ns3:PhysicalHousehold>
-</ns10:AccountTransferRequest>
```
Comment (4till2) [inline components/financial_assistance/spec/shared_examples/medicaid_gateway/Simple_Test_Case_E_New.xml:636]: Its a whitespace removal not the `AccountTransferReques`t tag, I'm not sure why it shows up in this way.

File: components/financial_assistance/spec/shared_examples/medicaid_gateway/Simple_Test_Case_E_New.xml [hunk 3]
```diff
@@ -145,6 +145,7 @@
             </ns5:SSFSignerAuthorizedRepresentativeAssociation>
             <ns5:SSFAttestation>
                 <ns5:SSFAttestationCollectionsAgreementIndicator>false</ns5:SSFAttestationCollectionsAgreementIndicator>
+                <ns5:SSFAttestationMedicaidObligationsIndicator>true</ns5:SSFAttestationMedicaidObligationsIndicator>
```
Comment (4till2) [inline components/financial_assistance/spec/shared_examples/medicaid_gateway/Simple_Test_Case_E_New.xml:148]: This is the actual addition.

**Review by erubinst — APPROVED** [review]
Looks good to me.  Test added in to test adding attestation to payload in enroll

**Review by mdkaraman — APPROVED** [review]
Nice work adding the spec coverage and improving the error logging!   Thanks for leaving the additional comments on the PR, as well. 🔥

**Pull request merged**